src/Sentry.Unity.Editor.iOS/SentryXcodeProject.cs

```diff
@@ -0,0 +1,129 @@
+using System.IO;
+using UnityEditor.iOS.Xcode;
+using UnityEditor.iOS.Xcode.Extensions;
+using UnityEngine;
+
+namespace Sentry.Unity.Editor.iOS
+{
+    internal class SentryXcodeProject
+    {
+        private const string FrameworkName = "Sentry.framework";
+
+        private const string MainPath = "MainApp/main.mm";
+        private const string OptionsPath = "MainApp/SentryOptions.m";
+
+        private readonly string _pathToProject;
+        private readonly PBXProject _project;
+        private readonly string _projectPath;
+
+        private readonly INativeMain _nativeMain;
+        private readonly ISentryNativeOptions _sentryNativeOptions;
+
+        private readonly string _relativeFrameworkPath;
+
+        internal SentryXcodeProject(
+            string pathToProject,
+            INativeMain? mainModifier = null,
+            ISentryNativeOptions? sentryNativeOptions = null)
+        {
+            _pathToProject = pathToProject;
+
+            _projectPath = PBXProject.GetPBXProjectPath(pathToProject);
+            _project = new PBXProject();
+
+            if (File.Exists(_projectPath))
+            {
+                _project.ReadFromString(File.ReadAllText(_projectPath));
+            }
+            else
+            {
+                Debug.LogWarning($"Could not locate generated Xcode project at {_projectPath}");
+            }
+
+            _relativeFrameworkPath = GetRelativeFrameworkPath(_pathToProject);
+
+            _nativeMain = mainModifier ?? new NativeMain();
+            _sentryNativeOptions = sentryNativeOptions ?? new NativeOptions();
+        }
+
+        public static SentryXcodeProject Open(string path)
+        {
+            return new (path);
+        }
+
+        public bool ValidateFramework()
+        {
+            if (string.IsNullOrEmpty(_relativeFrameworkPath))
+            {
+                Debug.LogWarning("Failed to locate the Sentry package within 'Frameworks'.");
+                return false;
+            }
+
+            var frameworkPath = Path.Combine(_pathToProject, _relativeFrameworkPath, FrameworkName);
+            if (!Directory.Exists(frameworkPath))
+            {
+                Debug.LogWarning($"Failed to locate 'Sentry.framework' within the Sentry package at '{frameworkPath}'.");
+                return false;
+            }
+
+            return true;
+        }
+
+        public void AddSentryFramework()
+        {
+            var targetGuid = _project.GetUnityMainTargetGuid();
+            var fileGuid = _project.AddFile(
+                Path.Combine(_relativeFrameworkPath, FrameworkName),
+                Path.Combine(_relativeFrameworkPath, FrameworkName));
```

Comment:
Save up some string allocations:
```suggestion
            var filePath = Path.Combine(_relativeFrameworkPath, FrameworkName);
            var fileGuid = _project.AddFile(filePath, filePath);
```